spec/lib/efile/nc/d400_calculator_spec.rb

```diff
@@ -87,6 +87,43 @@
     end
   end
 
+  describe "Line 12b: NCAGISubtraction" do
+    it "subtracts line 12a from line 8 (which is just fed agi)" do
+      allow(instance).to receive(:calculate_line_12a).and_return 10
+      allow_any_instance_of(DirectFileData).to receive(:fed_agi).and_return 100
+
+      instance.calculate
+      expect(instance.lines[:NCD400_LINE_12B].value).to eq 90
+    end
+  end
+
+  describe "Line 15: North Carolina Income Tax" do
+    context "value (line 14 (which = line 12b) * 0.045 rounded to the nearest dollar) is positive" do
+      it "returns the value rounded up" do
+        allow(instance).to receive(:calculate_line_12b).and_return 100
+
+        instance.calculate
+        expect(instance.lines[:NCD400_LINE_15].value).to eq 5
+      end
+
+      it "returns the value rounded down" do
```

Comment:
Im confused by the context in which it should be rounded down vs up, should these be in separate context blocks?

Reply:
oh sorry this description is probably confusing. it just means that it's always rounded (so not floored or rounded up or anything). it might be overkill to include both examples (so in the first case the number is 4.5 which rounds to 5 and in the other case it's 2.25 which rounds to 2), my thought was just that i could have accidentally implemented it as `.to_i` instead of `.round` and a test that only has the rounding-down case would not have caught that?